src/sentry/lang/native/minidump.py

```diff
@@ -96,6 +105,65 @@ def merge_process_state_event(data, state, cfi=None):
     data.setdefault('debug_meta', {})['images'] = images
 
 
+def merge_attached_event(mpack_event, data):
+    # Merge msgpack serialized event.
+    if sys.getsizeof(mpack_event) > MAX_MSGPACK_EVENT_SIZE_BYTES:
+        return
+
+    try:
+        event = msgpack.unpack(mpack_event)
+    except UnpackException as e:
+        minidumps_logger.exception(e)
+        return
+
+    for key in event:
+        value = event.get(key)
+        if value is not None:
+            data[key] = value
+
+
+def merge_attached_breadcrumbs(mpack_breadcrumbs, data):
+    # Merge msgpack breadcrumb file.
+    if sys.getsizeof(mpack_breadcrumbs) > MAX_MSGPACK_BREADCRUMB_SIZE_BYTES:
+        return
+
+    try:
+        unpacker = msgpack.Unpacker(mpack_breadcrumbs)
+    except UnpackException as e:
+        minidumps_logger.exception(e)
+        return
+
+    levels = {-1: 'debug', 0: 'info', 1: 'warning', 2: 'error', 3: 'critical'}
+    breadcrumbs = []
+    for crumb in unpacker:
```

Comment:
the try/except for `UnpackException` needs to go around the loop as the iteration can raise an `UnpackException` on bad data.